### ISSUE TITLE
Prediction

### ISSUE THREAD

**Issue by shunzhang** [shunzhang/RACHEL#6]
1 Split data into Training data and testing data
2 On the training set pick features
3 On the training set pick prediction function
4 On the training set cross validation
5 Apply to test data set and refine
## some useful code

lm1 <- lm(eruptions ~ waiting,data=trainFaith)
summary(lm1)
# Model fit

plot(trainFaith$waiting,trainFaith$eruptions,pch=19,col="blue",xlab="Waiting",ylab="Duration")
lines(trainFaith$waiting,lm1$fitted,lwd=3)
newdaya<- 
predict(lm1, newdata)
# plot predictions - raning and testing data

par(mfrow=c(1,2))
plot(trainFaith$waiting,trainFaith$eruptions,pch=19,col="blue",xlab="Waiting",ylab="Duration")
lines(trainFaith$waiting,predict(lm1),lwd=3)
plot(testFaith$waiting,testFaith$eruptions,pch=19,col="blue",xlab="Waiting",ylab="Duration")
lines(testFaith$waiting,predict(lm1,newdata=testFaith),lwd=3)
# getting training set/test set errors
# Calculate RMSE on training

sqrt(sum((lm1$fitted-trainFaith$eruptions)^2))
# Calculate RMSE on test

sqrt(sum((predict(lm1,newdata=testFaith)-testFaith$eruptions)^2))
# prediction intervals

pred1 <- predict(lm1,newdata=testFaith,interval="prediction")
ord <- order(testFaith$waiting)
plot(testFaith$waiting,testFaith$eruptions,pch=19,col="blue")
matlines(testFaith$waiting[ord],pred1[ord,],type="l",,col=c(1,2,2),lty = c(1,1,1), lwd=3)
